packages/ss58_codec/test/ss58_codec_test.dart

```
@@ -56,78 +56,178 @@ void main() {
+              exception.toString() == 'Bad Length Address.'),
+        ),
+      );
       expect(
-          () => SS58Codec.decode(
-              '3rr8zEMfiR2AjQYpVvncJCAcRP7CLzfxMGuUU2Pw2MCotrsQnS'),
-          throwsException);
+        () => SS58Codec.decode(
+            '4uqUGom7PszVSaZipgHYwVNsMESj1W6cmMZYeXGFeeXGX6VEqm'),
+        throwsA(
+          predicate((exception) =>
+              exception is BadAddressLengthException &&
+              exception.toString() == 'Bad Length Address.'),
+        ),
+      );
+    });
+
+    /// example of invalid decoded address.
+    test('invalid check sum exception', () {
       expect(
-          () => SS58Codec.decode(
-              '4uqUGom7PszVSaZipgHYwVNsMESj1W6cmMZYeXGFeeXGX6VEqm'),
-          throwsException);
+        () => SS58Codec.decode(
+            '3HX1zEyzCbxeXe34JY8SNSVAZ6djFccsV5f67PTied4CcWHs'),
+        throwsA(
+          predicate((exception) =>
+              exception is InvalidCheckSumException &&
+              exception.toString() == 'Invalid checksum'),
+        ),
+      );
     });
   });
 
-  // Encode Exception testing
+  /// [SS58Codec] `encode` method Exception testing
   group('encode: throw exception -> ', () {
+    /// [Address] with invalid `prefix`
     test('invalid prefix ', () {
+      /// [Address] invalid because of `negative` prefix
+      final addressWithNegativePrefix =
+          Address(prefix: -1, bytes: Uint8List.fromList([]));
       expect(
-          () => SS58Codec.encode(
-              Address(prefix: -1, bytes: Uint8List.fromList([]))),
-          throwsA(isA<InvalidPrefixException>()));
+        () => SS58Codec.encode(addressWithNegativePrefix),
+        throwsA(
+          predicate((exception) =>
+              exception is InvalidPrefixException &&
+              exception.toString() == 'Invalid SS58 prefix byte: -1.'),
+        ),
+      );
+
+      /// [Address] invalid because of prefix is greater than the limit
```

You should break this into multiple tests. Good unit tests usually fail for a single reason (that's very utopic, but nice to have). You can have ` test('it should throw InvalidPrefixException when an address has negative prefix'` and ` test('it should throw InvalidPrefixException when prefix is greater than the limit'`